src/main/java/org/openelisglobal/testconfiguration/controller/TestAddController.java

```diff
@@ -198,10 +198,10 @@ private List<TestSet> createTestSets(TestAddParams testAddParams) {
         if (numericResults) {
             lowValid = StringUtil.doubleWithInfinity(testAddParams.lowValid);
             highValid = StringUtil.doubleWithInfinity(testAddParams.highValid);
-            lowReportingRange = StringUtil.doubleWithInfinity(testAddParams.lowReportingRange);
-            highReportingRange = StringUtil.doubleWithInfinity(testAddParams.highReportingRange);
-            lowCriticalRange = StringUtil.doubleWithInfinity(testAddParams.lowCriticalRange);
-            highCriticalRange = StringUtil.doubleWithInfinity(testAddParams.highCriticalRange);
+            lowRangeLow = StringUtil.doubleWithInfinity(testAddParams.lowRangeLow);
```

Comment:
Is there a reason lowReportingRange and highReportingRange have been converted to lowRangeLow and highRangeHigh?

Personally I would omit passing the Doubles into createResultLimits and only pass in TestAddParams so all of the StringUtil.doubleWithInfinity(value) can be inside createResultLimits, with independant null safety checks on each pair (reportingRange pair, lowCriticalRange pair, etc) (NOTE! Confirm where these values are stored in the JSONObject some might be at the top level (ie obj.get("lowValid")) while others may be inside an array inside the object (ie obj.get("resultLimits")... limitArray.get(i).get("lowNormal")).